test/src/test/java/hudson/security/HudsonPrivateSecurityRealmFIPSTest.java

```diff
@@ -149,6 +148,87 @@ private static void userCreationWithJBCryptPasswordsStep(JenkinsRule j) throws E
                 is("The hashed password was hashed with an incorrect algorithm. Jenkins is expecting $PBKDF2"));
     }
 
+    @Test
+    public void validatePasswordLengthForFIPS() throws Throwable {
+        rjr.then(HudsonPrivateSecurityRealmFIPSTest::validatePasswordLengthForFIPSStep);
+    }
+
+    private static void validatePasswordLengthForFIPSStep(JenkinsRule j) throws Exception {
+        HudsonPrivateSecurityRealm securityRealm = new HudsonPrivateSecurityRealm(false, false, null);
+        j.jenkins.setSecurityRealm(securityRealm);
+
+        User u1 =  securityRealm.createAccount("test", "password");
+
+        WebClient wc = j.createWebClient();
+        wc.login("test","password");
+
+        HtmlPage configurePage = wc.goTo(u1.getUrl() + "/security/");
+        HtmlPasswordInput password1 = configurePage.getElementByName("user.password");
+        HtmlPasswordInput password2 = configurePage.getElementByName("user.password2");
+        //Should fail as the password length is <14 (In FIPS mode)
+        password1.setText("mockPassword");
+        password2.setText("mockPassword");
+
+        HtmlForm form = configurePage.getFormByName("config");
+        assertThrows(FailingHttpStatusCodeException.class, () -> {
+            j.submit(form);
+        });
```

Comment:
See the `userLoginAfterEnablingFIPS` method above in this test class. Probably you should check the log message as well. That method is doing

```
assertThat(getLogRecords(), hasItem(incorrectHashingLogEntry()));
```

Reply:
Apparently , there are no logs events during the test execution.

Reply:
there must be as a FormException is thrown. If you mean that `getLogRecords()` is returning nothing you might be missing the `setupLogRecorder();`